packages/contracts-bedrock/test/governance/ProposalValidator.t.sol

```diff
@@ -1236,3 +1267,218 @@ contract ProposalValidator_Initialize_Test is ProposalValidator_Init {
         );
     }
 }
+
+/// @title ProposalValidator_SubmitCouncilMemberElectionsProposal_Test
+/// @notice Happy path tests for submitCouncilMemberElectionsProposal function
+contract ProposalValidator_SubmitCouncilMemberElectionsProposal_Test is ProposalValidator_Init {
+    string proposalDescription;
+
+    event ProposalVotingModuleData(bytes32 indexed proposalHash, bytes encodedVotingModuleData);
+
+    function setUp() public override {
+        super.setUp();
+
+        // Set CouncilMemberElections to use the approval voting module
+        vm.prank(owner);
+        validator.setProposalTypeData(
+            ProposalValidator.ProposalType.CouncilMemberElections,
+            ProposalValidator.ProposalTypeData({
+                requiredApprovals: PROPOSAL_REQUIRED_APPROVALS,
+                proposalVotingModule: 2
+            })
+        );
+
+        proposalDescription = "Council Member Elections Q4 2024";
+    }
+
+    function testFuzz_submitCouncilMemberElectionsProposal_succeeds(
+        uint8 optionCount,
+        uint128 criteriaValue
+    ) public {
+        optionCount = uint8(bound(optionCount, 2, type(uint8).max)); // Minimum 2 options to have valid criteria < optionCount
+        criteriaValue = uint128(bound(criteriaValue, 1, optionCount - 1)); // Must be less than optionCount
+
+        // Create dynamic array of option descriptions based on option count
+        string[] memory optionDescriptions = new string[](optionCount);
+        for (uint256 i = 0; i < optionCount; i++) {
+            optionDescriptions[i] = string(abi.encodePacked("Candidate ", vm.toString(i)));
+        }
+
+        // Create attestation for the proposal
+        bytes32 attestationUid = _createAttestation(topDelegate_A, ProposalValidator.ProposalType.CouncilMemberElections);
+
+        // Calculate expected proposal hash
+        bytes memory votingModuleData = _constructCouncilElectionVotingModuleData(optionDescriptions, criteriaValue);
+        bytes32 expectedHash = validator.hashProposalWithModule(
+            makeAddr("approvalVotingModule"), votingModuleData, keccak256(bytes(proposalDescription))
+        );
+
+        // Mock proposalSnapshot to return 0 (proposal doesn't exist in governor)
+        _mockAndExpect(
+            address(governor),
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        vm.prank(topDelegate_A);
+        bytes32 proposalHash = validator.submitCouncilMemberElectionsProposal(
+            criteriaValue, optionDescriptions, proposalDescription, attestationUid
+        );
+
+        assertEq(proposalHash, expectedHash);
```

Comment:
need more assertions like the proposal data for the proposal is returning the correct data